zebra-network/src/address_book.rs

```diff
@@ -314,6 +325,33 @@ impl AddressBook {
         meta_addr
     }
 
+    /// Returns true if `updated` has a last_connection_state of `Responded`
+    /// and there is either:
+    /// - no previous addr with the updated.addr IP, or
+    /// - updated has a more recent last_response time than the existing [`MetaAddr`]
+    ///   for that IP in `most_recent_by_ip`.
+    fn should_update_most_recent_by_ip(&self, updated: MetaAddr) -> bool {
+        updated
+            .last_response()
+            .is_some_and(|updated_last_response| {
+                self.most_recent_by_ip
+                    .get(&updated.addr.ip())
+                    .map_or(true, |prev| {
+                        prev.last_response() < Some(updated_last_response)
+                    })
+            })
```

Comment:
`last_response()` will currently only return `Some(_)` when `last_connection_state` is `Responded`, but in case that changes:

```suggestion
        updated.last_connection_state == PeerAddrState::Responded
            && updated
                .last_response()
                .is_some_and(|updated_last_response| {
                    self.most_recent_by_ip
                        .get(&updated.addr.ip())
                        .map_or(true, |prev| {
                            prev.last_response() < Some(updated_last_response)
                        })
                })
```